### PR TITLE
update golang 1.18 and k8s to 0.24.2

### PR DESCRIPTION
**Pull request by ranakan19**

### CODE REVIEW

**Review by xcoulon — COMMENTED** [review]
could you also upgrade the `setup-go`, `cache` and `golangci/golangci-lint-action GitHub actions to v3?

**Comment by ranakan19** [comment]
> 41bbbd5

@xcoulon done in [41bbbd5](https://github.com/codeready-toolchain/toolchain-common/pull/280/commits/41bbbd5acc3c11824c1884b207d33ff1a1dede41)

**Review by xcoulon — APPROVED** [review]

**Comment by xcoulon** [comment]
> > [41bbbd5](https://github.com/codeready-toolchain/toolchain-common/commit/41bbbd5acc3c11824c1884b207d33ff1a1dede41)
> 
> @xcoulon done in [41bbbd5](https://github.com/codeready-toolchain/toolchain-common/pull/280/commits/41bbbd5acc3c11824c1884b207d33ff1a1dede41)

thanks @ranakan19 !

File: go.mod [hunk 1]
```diff
@@ -74,37 +68,32 @@ require (
...
 	k8s.io/kubectl v0.24.0 // indirect
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9 // indirect
-	sigs.k8s.io/controller-tools v0.9.0 // indirect
 	sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2 // indirect
-	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+replace github.com/codeready-toolchain/api v0.0.0-20221102132326-7a7f4eb137db => github.com/ranakan19/api v0.0.0-20221201091806-57fc665ce345
```
Comment (mfrancisc) [inline go.mod:99]: friendly reminder!

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**